metadata/metadata_test.go

```diff
@@ -385,5 +404,197 @@ func TestTargetFilesCustomField(t *testing.T) {
 	targets.Signed.Targets["testTarget"] = targetFile
 	targetsJSON, err := targets.ToBytes(false)
 	assert.NoError(t, err)
-	assert.Equal(t, []byte("{\"signed\":{\"_type\":\"targets\",\"expires\":\"2030-08-15T14:30:45.0000001Z\",\"spec_version\":\"1.0.31\",\"targets\":{\"testTarget\":{\"custom\":{\"test\":true},\"hashes\":{},\"length\":0}},\"version\":1},\"signatures\":[]}"), targetsJSON)
+	assert.Equal(t, exampleTargetsBytes, targetsJSON)
+}
+
+func TestFromBytes(t *testing.T) {
```

Comment:
What you should be testing here is that a generated metadata object that is based on a byte stream corresponds to what you would expect it to be - fields, values, etc.